xbmc/platform/linux/input/LibInputKeyboard.cpp

```diff
@@ -247,15 +332,50 @@ void CLibInputKeyboard::ProcessKey(libinput_event_keyboard *e)
   if (!m_ctx || !m_keymap || !m_state)
     return;
 
-  XBMC_Event event = {};
-
   const uint32_t xkbkey = libinput_event_keyboard_get_key(e) + 8;
+  const xkb_keysym_t keysym = xkb_state_key_get_one_sym(m_state.get(), xkbkey);
   const bool pressed = libinput_event_keyboard_get_key_state(e) == LIBINPUT_KEY_STATE_PRESSED;
-
-  event.type = pressed ? XBMC_KEYDOWN : XBMC_KEYUP;
   xkb_state_update_key(m_state.get(), xkbkey, pressed ? XKB_KEY_DOWN : XKB_KEY_UP);
 
-  const xkb_keysym_t keysym = xkb_state_key_get_one_sym(m_state.get(), xkbkey);
+  bool flushComposer{false};
+  if (pressed && SupportsKeyComposition())
+  {
+    xkb_compose_state_feed(m_composedState.get(), keysym);
+    const xkb_compose_status composeStatus = xkb_compose_state_get_status(m_composedState.get());
+    switch (composeStatus)
+    {
+      case XKB_COMPOSE_COMPOSING:
+      {
+        const std::optional<XBMCKey> xbmcKey = TranslateDeadKey(keysym);
+        if (xbmcKey)
+        {
+          NotifyKeyComposingEvent(XBMC_KEYCOMPOSING_COMPOSING, xbmcKey.value());
+        }
+        return;
+      }
+      case XKB_COMPOSE_COMPOSED:
+      {
+        flushComposer = true;
+        NotifyKeyComposingEvent(XBMC_KEYCOMPOSING_FINISHED, XBMCK_UNKNOWN);
+        break;
```

Comment:
This status is handled differently than the others, the other states return while this falls through. Does that mean multiple XBMC_events are going to be sent? Is that the correct behavior? Or do we just need to notify the composition is complete and reset and return?

Reply:
Ah I see that later we get the composed key as a Unicode key.